hypothesis-python/docs/explanation/domain.rst

```diff
@@ -0,0 +1,22 @@
+Domain and distribution
+=======================
+
+.. note::
+
+    This page is mainly for users who are familiar with other property-based testing libraries, and who expect a way to control the distribution of inputs in Hypothesis, via e.g. a scale parameter for size or a frequency parameter for relative probabilities.
+
+Hypothesis makes a distinction between the *domain* of a strategy, and the *distribution* of a strategy.
+
+* The domain is the set of inputs that should be possible to generate. For instance, in ``lists(integers())``, the domain is lists of integers. For other strategies, particularly those that use |st.composite| or |assume| in their definition, the domain might be more complex.
+* The distribution is the probability with which different elements in the domain should be generated. For ``lists(integers())``, should Hypothesis generate many small lists? Large lists? More positive or more negative numbers? etc.
+
+Hypothesis takes a philosophical stance that property-based testing libraries should be responsible for selecting the distribution; not the user. As an intentional design choice, Hypothesis therefore lets you control the domain of inputs to your test, but not the distribution.
+
+There are a few reasons for this:
+
+* Humans are pretty bad at choosing bug-finding distributions! Some bugs are "known unknowns": you had a good idea that some part of the code was buggy in a particular way. Others are "unknown unknowns": you had no idea that a certain kind of bug was even possible until you stumbled across it. Humans tend to overtune distributions for the former kind of bug, and not enough for the latter.
```

Comment:
Two sub-points:

- Strategies are also used in many different tests, and the ideal distribution is probably different for each.  That's part of the motivation for Hypofuzz's adaptive backend.
- To the extent that we think humans _can_ choose a distribution which finds bugs faster, we build that into Hypothesis directly, and evaluations show that this is a very strong baseline.

Reply:
added the first. im not sure we should claim the second bullet - at least not that strongly - given that we claim in this document (and I think both agree at a personal level) that some users *can* choose distributions that are better than our heuristics, simply due to domain expertise. I think people reading this page are disproportionately likely to actually be able to benefit from distribution control, and we should avoid making them sad by saying they are part of the generalized bucket of users that we think can't beat hypothesis